### DISCUSSION TITLE
Using exactly one OS Window and forcing new kitty commands to open a new tab?

### DISCUSSION THREAD

**Discussion by dcraig327** [kovidgoyal/kitty#7936] [Q&A] [answered]
How could I allow only a single `kitty` window and executing this:
1. `kitty --detach` would focus on the existing `kitty` window.
2. `kitty --detach newtab` would focus as above plus append a new tab.

Use case is to simplify Super+T in Gnome, providing a single window as I haven't a need for multiple tty os windows. I've searched in the existing closed issues, in the discussions, on the web, the documentation, and tried kitty-control all with no luck. I maybe searching for the wrong thing, this sounds like a question that would've been asked before.

Unclear if kitty can do 1. or 2. above. `kitty --single-instance` does use the existing PID, but it creates a new window. No luck with sessions.

**Comment by kovidgoyal** [comment, marked as answer]
https://sw.kovidgoyal.net/kitty/remote-control/

**Reply by dcraig327** [reply]
thx. wasn't initially clear.

this is the script to accomplish the requirements
saved in a script bound to a hotkey set in the linux distro should work, for those with bash of course. i imagine modifying this script for other operating systems wouldn't be too difficult.

i have seen asked a few times when researching how to do this, people wanting a single hotkey to open kitty. mostly it involves a new instance. understandable that can't be supported in kitty as global hotkeys are an os feature.

there's a similar question asked in #392 -- however I wanted todo everything using kitty and kitty alone and i'm prone to moving my terminal all around the place. which reminds me, i could set this up as a systemd user service, minimized at launch, which prevents sending all errors to `/dev/null` below.

anyway thx. i believe you understand the use case by now. hope this helps someone else.

```
#!/bin/bash

#focus existing kitty os window or create & focus if n/a
good_kitty () {
  kitten @ focus-window --to unix:/tmp/mykitty &> /dev/null
  if [ $? -ne 0 ]; then
    kitty --detach --directory="~" -o allow_remote_control=yes --listen-on unix:/tmp/mykitty
  fi
  kitten @ focus-window --to unix:/tmp/mykitty &> /dev/null
}

#focus new kitty tab in existing kitty os window
hello_kitty () {
  kitten @ launch --type=tab --to unix:/tmp/mykitty &> /dev/null
  if [ $? -eq 0 ]; then
    kitten @ focus-window --to unix:/tmp/mykitty &> /dev/null
  else
    good_kitty
  fi
}

hello_kitty
```

**Reply by llakala** [reply]
This did indeed help me, thank you!!

**Comment by fab6** [comment]
Hi,
Interesting approach. May I ask, if there is also an option to combine several instances later on to get multiple tabs instead multiple windows?
Thank you in advance!
Best Regards
Fab

**Reply by dcraig327** [reply]
> Hi, Interesting approach. May I ask, if there is also an option to combine several instances later on to get multiple tabs instead multiple windows? Thank you in advance! Best Regards Fab

Hi. That's what the bash script above does. Works well for my needs.

**Reply by fab6** [reply]
Ah, ok, I misunderstood and could not get it working as I am on macos; the `detach` option seems not to be available; though I am not sure about the other options yet. I will check, but maybe someone has a suggestion. 
Thank you!
